### DISCUSSION TITLE
[New Feature] Allow int variable reference with other numeric operands in a multiplicative expression

### DISCUSSION THREAD

**Discussion by rdulmina** [ballerina-platform/ballerina-lang#35518] [General]
Currently, int literal is allowed in multiplicative expression with `float` and `decimal`. ie the following are currently supported
```bal
public function main() {
    decimal a = 4.5;
    decimal b = a * 2;
    io:println(b); // prints 9.0

    decimal c = a / 2;
    io:println(c); // prints 2.25

    decimal d = a % 2;
    io:println(d); // prints 0.5
}
```

Similarly with the float. But this is not supported with `int` variable reference. ie the following will fail
```bal
public function main() {
    int intVarRef = 2;
    float a = 4.5;
    float b = a * intVarRef; // operator '*' not defined for 'float' and 'int'

    float c = a / intVarRef; // operator '*' not defined for 'float' and 'int'

    float d = a % intVarRef; // operator '*' not defined for 'float' and 'int'
}
```

We are going to add the support for int variable reference as well. Your input is much appreciated on this.

Link to spec issue: https://github.com/ballerina-platform/ballerina-spec/issues/261
Link to lang issue:  https://github.com/ballerina-platform/ballerina-lang/issues/35179

**Comment by gimantha** [comment]
Do we cover constants as well? for e.g.


```ballerina
const i = 8;

public function main() {
    float f = 56;
    float _ = f / d;  // operator '/' not defined for 'float' and 'int'
}
```

**Reply by rdulmina** [reply]
Yes this is covered for constants as well

**Comment by rdulmina** [comment]
Since we are not going to allow `int/float` and `int%float`, we need to change the error message accordingly since `operator '/' not defined for 'int' and 'float'` implies `float/int` is also not allowed. Few suggestions for the error message are

- operator '/' not defined for 'int'(dividend) and 'float' (divisor)
- operation int/float is not supported

@hasithaa @gimantha @MaryamZi @pcnfernando Your suggestions are highly appreciated on this
